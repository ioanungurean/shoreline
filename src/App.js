import React, { useEffect, useState } from "react";

import { RANGES } from "./constants";
import { fetchChartData } from "./api";

import Search from "./components/Search/Search";
import Chart from "./components/Chart/Chart";

import { StyledApp } from "./styles/StyledApp";
import GlobalStyle from "./styles/GlobalStyles";

const App = () => {
  const [symbol, setSymbol] = useState("");
  const [range, setRange] = useState(RANGES[0]);
  const [chartData, setChartData] = useState(null);

  const handleChartData = async () => {
    const formattedData = await fetchChartData(symbol, range);
    setChartData(formattedData);
  };

  useEffect(() => {
    if (symbol) {
      handleChartData();
    }
  }, [symbol, range]);

  const onSelect = (selection) => {
    setSymbol(selection.value);
  };

  return (
    <StyledApp>
      <GlobalStyle />
      <Search onSelect={onSelect} />
      {symbol && chartData && (
        <Chart
          symbol={symbol}
          data={chartData}
          ranges={RANGES}
          range={range}
          selectedRange={range}
          setRange={setRange}
        />
      )}
      {symbol}
    </StyledApp>
  );
};

export default App;
