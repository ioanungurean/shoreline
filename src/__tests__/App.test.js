import React from "react";
import { render } from "@testing-library/react";

import App from "../App";

describe("AlertCard", () => {
  it("Renders a card with a counter", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
