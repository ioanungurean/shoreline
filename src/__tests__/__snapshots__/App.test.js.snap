// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertCard Renders a card with a counter 1`] = `
.c0 {
  color: #61dbfb;
}

<div>
  <h1
    class="c0"
  >
    React App
  </h1>
</div>
`;
